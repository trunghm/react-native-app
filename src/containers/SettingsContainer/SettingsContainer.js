'use strict'

import React from 'react'
import {connect} from 'react-redux'
import SettingComponent from './views/index'
import {bindActionCreators} from 'redux'
import * as languageAction from '../../actions/languageAction';
import * as settingsActions from '../../actions/settingsActions';


// Define which part of the state we're passing to this component
const mapStateToProps = (state) => ({
  settingsState: state.settingsReducer.toJS()
})

// Define the actions this component may dispatch
const mapDispatchToProps = (dispatch) => {
  return {
    languageAction: bindActionCreators(languageAction, dispatch),
    settingsActions: bindActionCreators(settingsActions, dispatch),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SettingComponent)
