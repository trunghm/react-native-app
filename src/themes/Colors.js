export const DARK_MODE = "dark";
export const LIGHT_MODE = "light";


const common = {
  primaryColor: '#F4F4F4',
  colorD8: "#D8D8D8",
  color9B: "#9B9B9B",
  colorDE: "#DE2A21",
};

const appColors = {
  primaryBg: [`#FFFFFF`, `#000000`],
  primary: ['#1496bb', '#1496bb'],
  text: [`#333333`, `#FFFFFF`],
  buttonTitle: ["#FFFFFF", "#FFFFFF"],
  buttonBgDisable: ["#d2d2d2", "#d2d2d2"],
  avatarBg: ["#d9e2e7", "#d9e2e7"]
};

const colors = theme => {
  const themeColors = {};
  for (const key in appColors) {
    themeColors[key] = appColors[key][theme === DARK_MODE ? 1 : 0];
  }
  return {common: common, ...themeColors};
};

export default colors;
