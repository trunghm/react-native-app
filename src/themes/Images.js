export default {
  splash: require('./images/splash.png'),
  logo: require('./images/logo.png')
}
