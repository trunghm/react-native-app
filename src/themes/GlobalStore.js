
import  Colors,{ LIGHT_MODE } from './Colors';
export default class GlobalStore {
  static theme = LIGHT_MODE;
  static color = Colors(LIGHT_MODE);
  // static image = Image(LIGHT_MODE);
}
