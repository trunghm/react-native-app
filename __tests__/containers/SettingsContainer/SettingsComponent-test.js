/**
 * @format
 */

import React from 'react';
import {SettingComponent} from '../../../src/containers/SettingsContainer/SettingsContainer';

import {View, Text, Switch, Button} from 'react-native'
// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';
import {DARK_MODE, LIGHT_MODE} from "../../../src/themes/Colors";


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

jest.mock('react-native-gesture-handler', () => {});
jest.mock('@react-navigation/stack', () => {
  return {
    createAppContainer: jest.fn().mockReturnValue(function NavigationContainer(props) {return null;}),
    createDrawerNavigator: jest.fn(),
    createMaterialTopTabNavigator: jest.fn(),
    createStackNavigator: jest.fn(),
    StackActions: {
      push: jest.fn().mockImplementation(x => ({...x,  "type": "Navigation/PUSH"})),
      replace: jest.fn().mockImplementation(x => ({...x,  "type": "Navigation/REPLACE"})),
    },
    NavigationActions: {
      navigate: jest.fn().mockImplementation(x => x),
    }
  }
});


describe('Testing  SettingsComponent component', () => {

  const defaultProps = {
    settingsState: {theme:LIGHT_MODE},
    settingsActions:jest.fn(),
    languageAction:jest.fn(),
    navigation:jest.fn(),
    languageState: {name: 'vi'}
  };
  const darkProps = {
    ...defaultProps,
    settingsState: {theme:DARK_MODE},
    languageState: {name: 'en'}
  };
  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <SettingComponent {...defaultProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });


  it('renders as expected', () => {
    const wrapper = mount(
      <SettingComponent  {...defaultProps}  />
    );

    expect(wrapper).toMatchSnapshot();

    const witchMode = wrapper.find(Switch).first();
    expect(witchMode.prop('value')).toEqual(false);
    const witchLanguage = wrapper.find(Switch).last();
    expect(witchLanguage.prop('value')).toEqual(true);


  });

});
