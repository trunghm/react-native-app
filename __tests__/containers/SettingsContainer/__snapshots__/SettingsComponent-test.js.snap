// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  SettingsComponent component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  SettingsComponent component renders snapshot as expected 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
      }
    }
  >
    Settings
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
      }
    }
  >
    Change theme
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
        }
      }
    >
      Light mode
    </Text>
    <RCTSwitch
      accessibilityRole="button"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={false}
    />
    <Text
      style={
        Object {
          "color": "#333333",
        }
      }
    >
      dark mode
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#333333",
      }
    }
  >
    Change language
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
        }
      }
    >
      EN
    </Text>
    <RCTSwitch
      accessibilityRole="button"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={true}
    />
    <Text
      style={
        Object {
          "color": "#333333",
        }
      }
    >
      VI
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;
